paddle/framework/operator.cc

```diff
@@ -411,7 +411,34 @@ void OperatorWithKernel::Run(const Scope& scope,
                  expected_kernel_key);
   }
 
-  kernel_iter->second->Compute(ctx);
+  if (actual_kernel_key == expected_kernel_key) {
+    kernel_iter->second->Compute(ctx);
+  } else {
+    Scope& op_scope = scope.NewScope();
+    auto input_vars = this->InputVars();
+    for (auto var_name : input_vars) {
+      op_scope.Var(var_name);
+    }
+
+    // TODO(qijun) get appropriate DeviceContext from DeviceContext pool
+    platform::DeviceContext* trans_dev_ctx = nullptr;
+
+    // TODO(qijun) get appropriate DataTransformFn from global map
+    using DataTransformFn = std::function<void(
+        const Variable& in, Variable* out, platform::DeviceContext* ctx)>;
+    DataTransformFn trans_fun = nullptr;
+
+    for (auto var_name : input_vars) {
```

Comment:
here has a problem that maybe not all the input vars need to be transformed

Reply:
I have not think out an elegant solution. Maybe we can make some hard codes before make data transform, just like

```
auto input_vars = this->InputVars();
if (op_type == "blabla") {
   input_vars.erase(...);
} else if () {
   ...
}

```